### PR TITLE
Rename IPomFacade to IMavenExecutableLocation

### PR DESCRIPTION
**Pull request by mickaelistria**
and only keep dereferenced methods.

### CODE REVIEW

File: org.eclipse.m2e.core/src/org/eclipse/m2e/core/embedder/IMavenExecutableLocation.java [hunk 1]
```diff
@@ -13,25 +13,16 @@
...
 
-import org.eclipse.core.resources.IFile;
-
 
 /**
- * The {@link IPomFacade} is a wrapper over the maven representation of a pom that represents something that could be
- * executed as a maven execution. You can expect it to do something similar to
- * <code>mvn -f {@link IPomFacade#getPomFile()}</code> on the commandline.
+ * The {@link IMavenExecutableLocation} is a wrapper over the maven representation of a pom that represents something
+ * that could be executed as a maven execution. You can expect it to do something similar toz
```
Comment (laeubi) [inline org.eclipse.m2e.core/src/org/eclipse/m2e/core/embedder/IMavenExecutableLocation.java:19]:
```suggestion
 * that could be executed as a maven execution. You can expect it to do something similar to
```

**Review by laeubi — APPROVED** [review]
Maybe we can even strip the "location" part here and just use `IMavenExecutable` but in the end I think the exact name does not matter much so I'm fine with both.

**Comment by mickaelistria** [comment]
The reason of test failures is

```
java.io.InvalidClassException: org.eclipse.m2e.core.internal.project.registry.MavenProjectFacade; no valid constructor
	at java.base/java.io.ObjectStreamClass$ExceptionInfo.newInvalidClassException(ObjectStreamClass.java:174)
	at java.base/java.io.ObjectStreamClass.checkDeserialize(ObjectStreamClass.java:921)
	at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2260)
	at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1768)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:543)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:501)
	at java.base/java.util.HashMap.readObject(HashMap.java:1552)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:577)
	at java.base/java.io.ObjectStreamClass.invokeReadObject(ObjectStreamClass.java:1231)
	at java.base/java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2458)
	at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2292)
	at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1768)
	at java.base/java.io.ObjectInputStream$FieldValues.<init>(ObjectInputStream.java:2641)
	at java.base/java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2492)
	at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2292)
	at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1768)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:543)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:501)
	at org.eclipse.m2e.core.internal.project.registry.ProjectRegistryReader.readWorkspaceState(ProjectRegistryReader.java:71)
	at org.eclipse.m2e.tests.MutableProjectRegistryTest.testForeignClassesInSerializedProjectRegistry(MutableProjectRegistryTest.java:254)
```

**Comment by laeubi** [comment]
MavenProjectFacade is a serializable class and you now extend a non serializable class instead of implementing just the interface, I don't think we save much by let `MavenProjectFacade extend ResourcePomFacade`.

**Comment by mickaelistria** [comment]
> I don't think we save much by let MavenProjectFacade extend ResourcePomFacade.

Agree.

**Comment by mickaelistria** [comment]
> Actually I try to avoid java object serialization whenever possible :-)

Yes. I'm also unsure what's intended with this seriarlization. I guess it's a way to not re-compute dependency tree on every restart or something of that sort. We'll figure out later.

**Comment by laeubi** [comment]
@mickaelistria can you resolve the conflict an we then merge this (so other change do not conflict with this?)

**Comment by mickaelistria** [comment]
Same failures as master, so most likely no regression, let's merge.

**Pull request merged**